src/common/telemetry/libjson/TelemetryJson.c

```diff
@@ -0,0 +1,432 @@
+            free(path);
+        }
+        exit(errno);
+    }
+    // Parent process continues immediately without waiting
+}
+
+// C interface implementations
+int OSConfigTelemetryOpen(OsConfigLogHandle log)
+{
+    // If already open, return success
+    if (g_telemetryLogger != NULL && g_telemetryLogger->isOpen)
+    {
+        return 0;
+    }
+
+    // Clean up any existing instance
+    if (g_telemetryLogger != NULL)
+    {
+        OSConfigTelemetryClose();
+    }
+
+    g_telemetryLogger = malloc(sizeof(struct TelemetryLogger));
+    if (!g_telemetryLogger)
+    {
+        OsConfigLogError(log, "Failed to allocate memory for TelemetryLogger");
+        return -1;
+    }
+
+    // Initialize the logger
+    g_telemetryLogger->logFile = NULL;
+    g_telemetryLogger->filename = NULL;
+    g_telemetryLogger->binaryDirectory = NULL;
+    g_telemetryLogger->isOpen = 0;
+    g_telemetryLogger->log = log;
+
+    // Generate filename
+    g_telemetryLogger->filename = generateRandomFilename();
+    if (!g_telemetryLogger->filename)
+    {
+        free(g_telemetryLogger);
+        g_telemetryLogger = NULL;
+        OsConfigLogError(log, "Failed to generate random filename for telemetry log");
+        return -1;
+    }
+
+    // Open file
+    g_telemetryLogger->logFile = fopen(g_telemetryLogger->filename, "a");
+    if (!g_telemetryLogger->logFile)
+    {
+        free(g_telemetryLogger->filename);
+        free(g_telemetryLogger);
+        g_telemetryLogger = NULL;
+        OsConfigLogError(log, "Failed to open telemetry log file: %s", strerror(errno));
+        return -1;
+    }
+
+    g_telemetryLogger->isOpen = 1;
+    return 0;
+}
+
+int OSConfigTelemetryClose(void)
+{
+    if (g_telemetryLogger == NULL)
+    {
+        return -1;
+    }
+
+    if (!g_telemetryLogger->isOpen)
+    {
+        OsConfigLogError(g_telemetryLogger->log, "Telemetry logger is not open");
+        return -1;
+    }
+
+    if (g_telemetryLogger->logFile)
+    {
+        fclose(g_telemetryLogger->logFile);
+        g_telemetryLogger->logFile = NULL;
+    }
+
+    g_telemetryLogger->isOpen = 0;
+
+    // Set binary directory if not already set
+    if (!g_telemetryLogger->binaryDirectory)
+    {
+        g_telemetryLogger->binaryDirectory = getModuleDirectory();
+    }
+
+    // Run telemetry proxy if binary directory is available
+    if (g_telemetryLogger->binaryDirectory && g_telemetryLogger->filename)
+    {
+        runTelemetryProxy(g_telemetryLogger->filename, g_telemetryLogger->binaryDirectory);
+    }
+
+    // Clean up
+    if (g_telemetryLogger->filename)
+    {
+        free(g_telemetryLogger->filename);
+    }
+    if (g_telemetryLogger->binaryDirectory)
+    {
+        free(g_telemetryLogger->binaryDirectory);
+    }
+    free(g_telemetryLogger);
+    g_telemetryLogger = NULL;
+
+    return 0;
+}
+
+int OSConfigTelemetryLogEvent(const char* eventName, const char** keyValuePairs, int pairCount)
```

Comment:
>OSConfigTelemetryLogEvent [](http://example.com/codeflow?start=4&length=25)

Can we move most of this C code as C++ code into the Telemetry Client executable? And leave in C here only a set of macros for writing lines to the file?